.circleci/config.yml

```diff
@@ -169,7 +171,8 @@ jobs:
               python -m venv venv
               source venv/bin/activate
               make setup
-              make pipenv-lock
+              pip install --upgrade pip
+              make pipenv-install
```

Comment:
`--system` tells `pipenv` to use the "system" version of `pip` rather than the one in the virtual environment. I'm not sure what this was useful for, and I can't think of why it would have been preferred. In any case I've change this so we upgrade the environment local pip right above this, so  we should be running from latest.

Reply:
`--ignore-pipfile` is telling it to use just what's in the lockfile, and ignore the actual `lib/Pipfile`. This is no longer appropriate since we've discarded the use of lockfiles.

Same for the next question.

Reply:
I think `--system` is useful for docker images since we typically don't need to install a venv into a python based image. Are we using a venv in circleci? How about GKE?

Reply:
We install a venv here to concentrate our dependency installs into a directory that CircleCI can then cache for us. I'm not sure why `--system` was passed before or what precisely it did, but we do want these package installs to be in the `venv/` directory based on all the rest of the code in this file that caches and restores that directory.

This file is just for CircleCI, so yes, we're using a venv there. Can you expand on the GKE question? With S4T, we are not using a virtualenv (but are doing `--user` installs right now), but the `pipenv-lock` make target was only ever used to support CircleCI, I believe.